### DISCUSSION TITLE
Route handler: How to append userId inside request on middlewares

### DISCUSSION THREAD

**Discussion by aguirrealvaro** [vercel/next.js#52432] [App Router] [answered]
Hey, I have a middleware that works as authenticator to protect some routes:

```
export async function middleware(req: NextRequest) {
  const res = NextResponse.next();

  const { session } = await getSession();

  if (!session) {
    return NextResponse.json("Not auth", { status: 400 });
  }

  // TO DO: append session.user.id to request

  NextResponse.next();
}
```

I would like to append inside the request object the `userId` of the current logged user, something like:

`req.userId = session.user.id`

In order to have the userId value inside every protected route

Thanks.

**Comment by robertpiosik** [comment, marked as answer]
What you want to do is set up a cookie and read it whenever you need to know `userId`.

**Reply by aguirrealvaro** [reply]
Thanks!
